app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,42 @@
 class PostsController < ApplicationController
-  def index; end
+  before_action :find_post, only: [:show]
+  before_action :initialize_like
 
-  def show; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: params[:user_id])
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build
+  end
+
+  def create
+    puts 'i am here pushkar'
```

Comment:
- [ ] [OPTIONAL] Kindly remove every log statement on the code base because they do slow down the code base and it is not also a good practice to leave console statements on your code.